lib/nerves_init_gadget/network_manager.ex

```diff
@@ -19,7 +19,13 @@ defmodule Nerves.InitGadget.NetworkManager do
     SystemRegistry.register()
 
     # Initialize networking
-    Nerves.Network.setup(opts.ifname, ipv4_address_method: opts.address_method)
+    network_opts =
+      :nerves_network
+      |> Application.get_env(:default, [])
+      |> Keyword.get(to_atom(opts.ifname), [])
+      |> Keyword.put(:ipv4_address_method, opts.address_method)
+
+    Nerves.Network.setup(opts.ifname, network_opts)
```

Comment:
Since we are just using Nerves.Network can't we just remove all this altogether?

Reply:
The reason this is here at all is that `nerves_init_gadget` has its own `ifname` and `address_method` settings you can configure, outside of the `nerves_network` configuration. This could all go away if we didn't have those options in `nerves_init_gadget` and just instructed people to configure `nerves_network` directly.

I'd personally be fine with that because I think it would be less-confusing, but I didn't want to break compatibility with this change unless that's the direction we've decided we want to go.

Reply:
I like that idea as well. I'd like to hold off until after ElixirConf on any backwards incompatible changes, so let's revisit then.